### ISSUE TITLE
Symfony 4.4.17: Segmentation fault with PHPUnit 9.5.0 when using http_basic authentication

### ISSUE THREAD

**Issue by sdespont** [symfony/symfony#39479]
**Symfony version(s) affected**: 4.4.17

**PHP Version:** 7.4.13

**Description**  
When using a `http_basic `authentication, PHPunit crashes with a segmentation fault when `$this->client->request(...)` is called

```
vagrant@homestead:~/vendor$ bash /www/prod/bin/phpunit --debug --verbose --configuration /www/prod/phpunit.xml /www/prod/tests/Navigation/SecurityAccess/SecurityAccessDemoTest.php --filter testViewerAccess
PHPUnit 9.5.0 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.4.13
Configuration: /www/prod/phpunit.xml

Testing App\Tests\Navigation\SecurityAccess\SecurityAccessDemoTest
Test 'App\Tests\Navigation\SecurityAccess\SecurityAccessDemoTest::testViewerAccess' started
/www/prod/bin/phpunit: line 10: 16439 Segmentation fault      (core dumped) "${dir}/phpunit" "$@"
```

**How to reproduce**  

config/packages/test/security.yaml

```
security:
    firewalls:
        main:
            http_basic: ~
```

```
    public function testViewerAccess()
    {
        // Create a new client to browse the application with ROLE_USER
        $this->client = static::createClient(array(), array(
            'PHP_AUTH_USER' => 'UserTest,
            'PHP_AUTH_PW'   => '123',
        ));
        $this->client->request('GET', '/');
    }
```

**Possible Solution**  
No idea

**Additional context**  
Theses tests works well with Symfony 3.4. I have upgraded to 4.4 this week.
OPCache is disabled, XDebug is disabled

![image](https://user-images.githubusercontent.com/646632/102008178-f6014380-3d2e-11eb-96b7-922f642a638b.png)

Docs : https://symfony.com/doc/4.4/testing/http_authentication.html

**Comment by chalasr** [comment]
The details you provide are not sufficient. Please create a minimal application containing enough code to reproduce the issue.

**Comment by michaljusiega** [comment]
@sdespont 

Segfault are PHP bugs, not Symfony. For more details use `gdb`:

`gdb php`
then run into interactive mode:
`/www/prod/bin/phpunit --debug --verbose --configuration /www/prod/phpunit.xml /www/prod/tests/Navigation/SecurityAccess/SecurityAccessDemoTest.php --filter testViewerAccess`

**Comment by BR0kEN-** [comment]
I noticed it happens when `phpunit.xml` has `<env name="XDEBUG_MODE" value="coverage"/>` however `XDEBUG_MODE=coverage ./bin/phpunit` works.

**Comment by sdespont** [comment]
@BR0kEN- @michaljusiega Thanks for your support.

I don't have XDebug enabled, neither the `phpunit.xml ` code you wrote above inside.

I have tested with PHP7.3.23 and the same problem occurs.

Also, it happens **ONLY** when the login is a success. If I set false login for example, the `$this->client->request('GET', '/');` returns the login page as expected. But if the login and the password are rights, the segfault error happens.

I am using a vagrant box with homestead BTW. Tested with homestead 10.15.0 and 10.17.0.

In summary, not related to PHP version, homestead, XDebug

**Comment by sdespont** [comment]
Ok, that was a tricky one!

My project is using Monolog, and I add some extra details for debug information. The service is defined like this :

    monolog.processor.user:
        class: App\Services\ToolBox\Monolog\ExtraProcessor
        arguments:  ["@security.token_storage"]
        tags:
            - { name: monolog.processor }
            - { name: kernel.event_listener, event: kernel.request, method: onKernelRequest }

The service is invoked every times and add some extra details about the user. 

    /**
     * @param array $record
     * @return array
     */
    public function __invoke(array $record)
    {
        if (null !== $this->user) {
            $record['extra']['user'] = $this->user->getUserDetails($record['level_name'] === 'INFO' ? false : true);
        }
        
        if (null !== $this->postParams) {
            $record['extra']['postParams'] = $this->postParams;
        }
        
        return $record;
    }
          

The tricky parts was in the `getUserDetails` method:

    /**
     * @param bool $addUserGroupDetails
     * @return string
     */
    public function getUserDetails(bool $addUserGroupDetails)
    {
        $roles = "";
        foreach ($this->roles as $role) {
            $roles .= $role.", ";
        }

        if ($addUserGroupDetails) {
            $userGroups = "";
            foreach ($this->link_user_to_user_groups as $link) {
                /* @var LinkUserToUserGroup $link */
                $userGroup = $link->getUserGroup();
                $userGroups .= $userGroup->getNameAndType().", ";
            }
        } else {
            $userGroups = "N/A";
        }

        return 'Login: '.$this->username.' (ID:'.$this->id.') as '.$this->firstname." ".$this->lastname
            ." / Email[".$this->email."]"
            ." / Roles[".rtrim($roles, ", ")."]"
            ." / User Groups[".rtrim($userGroups, ", ")."]"
        ;
    }

No rocket science here, but the method is using a collection.... which was loaded with `LAZY `option. **Switching the option to EAGER solve the problem!**

    <one-to-many field="link_user_to_user_groups" target-entity="App\Entity\TableLink\LinkUserToUserGroup" mapped-by="user" fetch="EAGER"/>

Nice idea to find segfault BTW : https://gist.github.com/lyrixx/17074868cdfabd4c783e

I don't know if the problem is related to Doctrine or Monolog, is there someone able to explain why it was throwing a PHP segfault error?

**Comment by xabbuh** [comment]
Let's close here as the issue doesn't seem related to Symfony.
